src/dmd/cppmangle.d

```diff
@@ -150,18 +150,21 @@ private final class CppMangleVisitor : Visitor
     Objects components;         // array of components available for substitution
     OutBuffer* buf;             // append the mangling to buf[]
     Loc loc;                    // location for use in error messages
+    Strings* abiTags;           // record of the processed abi-tags
```

Comment:
Can we use D arrays?

Reply:
DMD arrays mange their own memory.

Reply:
@SSoulaimane DMD is a mechanical translation from a C++ code base so it contains a lot of code that is not considered idiomatic D.  Please prefer modern idiomatic D types and patterns for any new code so we can gradually transition DMD away from the cruft of the past.

Reply:
Please use Ddoc comments.

Reply:
The array is being filled by the visitor, I'm not sure that using D arrays is an improvement here because they rely on the GC.

Reply:
> As I understand DMD is moving away from the GC.

No, not that I've heard of.

Reply:
> No, not that I've heard of.

I thought using `mem.xmalloc` and `mem.xfree` was supposed to make DMD more responsible about memory. 

> Also, D arrays can be allocated using other means than the GC.

How do you append to it.

The array fed to the visitor is stack allocated hence automatically freed.